gender_novels/corpus_gen.py

```diff
@@ -500,7 +557,10 @@ def get_author_gender_wikidata(author):
     """
     try:
         match = re.match(AUTHOR_NAME_REGEX, author)
-        author_formatted = match.groupdict()['first_name'] + " " + match.groupdict()['last_name']
+        first_name = match.groupdict()['first_name']
+        if (match.groupdict()['real_name'] != None):
+            first_name = match.groupdict()['real_name']
+        author_formatted = first_name + " " + match.groupdict()['last_name']
         if (match.groupdict()['suffix'] != None):
             author_formatted += match.groupdict()['suffix']
     except (TypeError, AttributeError):
```

Comment:
This comment actually applies to the `dictionary = item.get()` or `clm_dict = dictionary['claims']` line and below: another good opportunity for factoring out to a small routine: `get_gender_from_wiki_claims(clm_dict)`